src/schema/resolvers/user.js

```diff
@@ -1,26 +1,59 @@
-const { v4: uuidv4 } = require('uuid');
+const { v4: uuidv4 } = require('uuid')
+const jwt = require('jsonwebtoken')
 
 export default {
-    Query: {
-        users: (parent, args, context, info) => context.models.user.findAll(),
-        user: (parent, args, context, info) => context.models.user.findByPk(args.id)
+  Query: {
+    users: (parent, args, context, info) => context.models.user.findAll(),
+    user: (parent, args, context, info) => context.models.user.findByPk(args.id),
+    currentUser: (parent, args, context, info) => context.user
+  },
+  Mutation: {
+    createUser: async (parent, args, context, info) => {
+      return createUserDB(args.input, context)
     },
-    Mutation: {
-        createUser: async (parent, args, context, info) => {
-            await console.log(context.models.user)
-            await console.log(args.input) // por alguna razon no tiene datos aunque le pase todo un inputUser
+    signup: async (parent, args, context, info) => {
 
-            var saltValue = context.models.user.getRandomSalt()
-            return context.models.user.create({
-                id: uuidv4(),
-                firstName: args.input.firstName,
-                lastName: args.input.lastName,
-                username: args.input.username,
-                salt : saltValue,
-                password : context.models.user.encryptPassword(args.input.password, saltValue)
-            })
-            //context.models.user.upsert
+      let newUser = await context.models.user.findOne({ where: { username: args.data.username } })
+      let error = 'user already exists'
+      if (!newUser) {
+        newUser = await createUserDB(args.data, context)
+        error = null
+      }
+      const { id, username } = newUser
+      const jwtUser = jwt.sign({ id, username }, process.env.SECRET, { expiresIn: '1d', subject: id }) 
+
+      return { user: newUser, jwt: jwtUser, authError: error }
+    },
+    signin: async (parent, args, context, info) => {
+
+      const newUser = await context.models.user.findOne({ where: { username: args.data.username } })
+      let error = null
+      if (!newUser) {
+        error = 'user doesnt exists'
+        return { user: null, jwt: null, authError: error }
+      } else {
+        if (!newUser.passwordMatches(args.data.password)) {
+          error = 'wrong password'
+          return { user: null, jwt: null, authError: error }
         }
+      }
```

Comment:
The error handling is ok, but you could write it even shorter and readable:
```suggestion
if (!newUser) return { user: null, jwt: null, authError: 'user doesnt exists' }
if (!newUser.passwordMatches(args.data.password)) return { user: null, jwt: null, authError: 'wrong password' }
```